walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,102 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        // Get necessary info from the MeetingRequest object.
+        Collection<String> requiredAttendees = request.getAttendees();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+        long duration = request.getDuration();
+
+        // Get all the timeRanges where the attendees have scheduled meetings.
+        ArrayList<TimeRange> requiredTimes = getEventTimes(events, requiredAttendees);
```

Comment:
maybe this should be called "unvailableTimesForRequiredAttendees"? and similar for optionalTimes?

Reply:
Then allTimes => unavailableTimesForAllAttendees?